src/main/navigation/navigation.c

```diff
@@ -2022,11 +2035,20 @@ static navigationFSMEvent_t navOnEnteringState_NAV_STATE_WAYPOINT_RTH_LAND(navig
 {
     UNUSED(previousState);
 
+#ifdef USE_FW_AUTOLAND
+    if (posControl.fwLandState.landAborted) {
+        return NAV_FSM_EVENT_SWITCH_TO_WAYPOINT_FINISHED;
+    }
+#endif
+
     const navigationFSMEvent_t landEvent = navOnEnteringState_NAV_STATE_RTH_LANDING(previousState);
 
+#ifdef USE_FW_AUTOLAND
     if (landEvent == NAV_FSM_EVENT_SWITCH_TO_NAV_STATE_FW_LANDING) {
         return NAV_FSM_EVENT_SWITCH_TO_NAV_STATE_FW_LANDING;
-    } else if (landEvent == NAV_FSM_EVENT_SWITCH_TO_RTH_HOVER_ABOVE_HOME) {
+    } else 
+#endif
+    if (landEvent == NAV_FSM_EVENT_SWITCH_TO_RTH_HOVER_ABOVE_HOME) {
```

Comment:
Can this ever be returned from `navOnEnteringState_NAV_STATE_RTH_LANDING `?

Reply:
Yes, occurs at landing waypoints.

Reply:
But it isn't specifically returned from `navOnEnteringState_NAV_STATE_RTH_LANDING `is my point so `landEvent `can never have this value surely ?

It can though have a value of `NAV_FSM_EVENT_SWITCH_TO_EMERGENCY_LANDING` which isn't handled by the WP RTH landing state. But that's an existing error so nothing to do with Auto Landing.

Would probably be better to have RTH/WP landing handled as a separate common end phase of either mode ... a future change though to tidy things up.

Reply:
Oh, you mean `NAV_FSM_EVENT_SWITCH_TO_RTH_HOVER_ABOVE_HOME` sorry. 
Yes, you're right.

Reply:
Sorry, yes I meant `NAV_FSM_EVENT_SWITCH_TO_RTH_HOVER_ABOVE_HOME`. Swap it for `NAV_FSM_EVENT_SWITCH_TO_EMERGENCY_LANDING `and it'll fix 2 issues at the same time.